diff --git a/QSSI.14/system/core/rootdir/init.rc b/QSSI.14/system/core/rootdir/init.rc
index 5ec96ea..a8caba3 100644
--- a/QSSI.14/system/core/rootdir/init.rc
+++ b/QSSI.14/system/core/rootdir/init.rc
@@ -1358,7 +1358,7 @@ on userspace-reboot-fs-remount
   umount /data_mirror/ref_profiles
   umount /data_mirror
   remount_userdata
-  start bootanim
+#  start bootanim
 
 on userspace-reboot-resume
   trigger userspace-reboot-fs-remount
@@ -1381,3 +1381,5 @@ on property:persist.device_config.mglru_native.lru_gen_config=core_and_nonleaf_y
   write /sys/kernel/mm/lru_gen/enabled 5
 on property:persist.device_config.mglru_native.lru_gen_config=all
   write /sys/kernel/mm/lru_gen/enabled 7
+
+import /system/etc/init/init.smartnfc.rc
\ No newline at end of file
