diff --git a/QSSI.14/hardware/nxp/secure_element/snxxx/1.2/Android.bp b/QSSI.14/hardware/nxp/secure_element/snxxx/1.2/Android.bp
index 5ca335d..ba129a4 100644
--- a/QSSI.14/hardware/nxp/secure_element/snxxx/1.2/Android.bp
+++ b/QSSI.14/hardware/nxp/secure_element/snxxx/1.2/Android.bp
@@ -54,7 +54,7 @@ cc_binary {
         "libchrome",
         "vendor.nxp.nxpese@1.0",
         "vendor.nxp.nxpnfc@2.0",
-        "vendor.nxp.nxpnfc_aidl-V1-ndk",
+        //"vendor.nxp.nxpnfc_aidl-V1-ndk",
         "android.hardware.nfc@1.0",
         "android.hardware.nfc@1.1",
         "android.hardware.nfc@1.2",
diff --git a/QSSI.14/hardware/nxp/secure_element/snxxx/aidl/Android.bp b/QSSI.14/hardware/nxp/secure_element/snxxx/aidl/Android.bp
index 61eca51..c4e397e 100644
--- a/QSSI.14/hardware/nxp/secure_element/snxxx/aidl/Android.bp
+++ b/QSSI.14/hardware/nxp/secure_element/snxxx/aidl/Android.bp
@@ -46,10 +46,12 @@ cc_binary {
         "libmemunreachable",
         "libutils",
         "ese_spi_nxp_snxxx",
-        "nfc_nci_nxp_snxxx",
+        //"nfc_nci_nxp_snxxx",
+        "nfc_nci_nxp_pn72xx",
         "android.hardware.nfc@1.2",
         "android.hardware.nfc-V1-ndk",
-        "vendor.nxp.nxpnfc_aidl-V1-ndk",
+        //"vendor.nxp.nxpnfc_aidl-V1-ndk",
+        //"aidl.vendor.nxp.nxpnfc_aidl-V1-ndk"
     ],
 
     include_dirs: [
diff --git a/QSSI.14/hardware/nxp/secure_element/snxxx/libese-spi/Android.bp b/QSSI.14/hardware/nxp/secure_element/snxxx/libese-spi/Android.bp
index c45dd01..97bb163 100755
--- a/QSSI.14/hardware/nxp/secure_element/snxxx/libese-spi/Android.bp
+++ b/QSSI.14/hardware/nxp/secure_element/snxxx/libese-spi/Android.bp
@@ -44,7 +44,7 @@ cc_library_shared {
         "src/include",
     ],
     include_dirs: [
-        "hardware/nxp/nfc/snxxx/extns/impl/nxpnfc/aidl",
+        //"hardware/nxp/nfc/snxxx/extns/impl/nxpnfc/aidl",
         "hardware/nxp/nfc/snxxx/extns/impl/nxpnfc/2.0",
         "hardware/nxp/secure_element/snxxx/extns/impl",
         "hardware/nxp/secure_element/snxxx/ese-clients/inc"
@@ -79,7 +79,8 @@ cc_library_shared {
         "libchrome",
         "vendor.nxp.nxpese@1.0",
         "vendor.nxp.nxpnfc@2.0",
-        "vendor.nxp.nxpnfc_aidl-V1-ndk",
+        //"vendor.nxp.nxpnfc_aidl-V1-ndk",
+        //"aidl.vendor.nxp.nxpnfc_aidl-V1-ndk"
     ],
 
     product_variables: {
